internal/webhook/external/bitbucket.go

```diff
@@ -155,7 +149,88 @@ func (b *bitbucketWebhookReceiver) getHandler(requestBody []byte) http.HandlerFu
 
 		logger = logger.WithValues("repoURL", repoURL)
 		ctx = logging.ContextWithLogger(ctx, logger)
-
-		refreshWarehouses(ctx, w, b.client, b.project, repoURL)
+		rc := &refreshEligibilityChecker{
+			newCode: newBitBucketCodeChange(payload),
+		}
+		refreshWarehouses(ctx, w, b.client, b.project, rc, repoURL)
 	})
 }
+
+type bitBucketPushEvent struct {
+	Actor struct {
+		Name         string `json:"name"`
+		EmailAddress string `json:"emailAddress"`
+	} `json:"actor"`
+	Push struct {
+		Changes []struct {
+			New struct {
+				Name   string `json:"name"`
+				Target struct {
+					Hash    string `json:"hash"`
+					Message string `json:"message"`
+					Date    string `json:"date"`
+				} `json:"target"`
+			} `json:"new"`
+		} `json:"changes"`
+	} `json:"push"`
+	Repository struct {
+		Links struct {
+			HTML struct {
+				Href string `json:"href"`
+			} `json:"html"`
+		} `json:"links"`
+	} `json:"repository"`
+}
+
+// newBitBucketCodeChange creates a new codeChange instance from a Bitbucket push event.
+// It extracts the metadata from the event payload. This is used downstream to determine which Warehouses
+// should be refreshed in response to the event based on the commit selection
+// strategy configured for the Warehouse.
+//
+// See the Bitbucket Push event payload documentation for more details:
+//
+//	https://support.atlassian.com/bitbucket-cloud/docs/event-payloads/#Push
+func newBitBucketCodeChange(e *bitBucketPushEvent) *codeChange {
+	hc := e.Push.Changes[0].New
```

Comment:
What does `hc` stand for? After looking at this code for quite some time, I'm still not sure. A more descriptive name would probably add quite a bit of clarity. All the other receivers have a variable with this name as well.